### PR TITLE
Support for Certified Data Sources in the REST API

### PR DESCRIPTION
**Pull request by t8y8**

### CODE REVIEW

File: tableauserverclient/models/datasource_item.py [hunk 1]
```diff
@@ -104,22 +130,30 @@ def from_response(cls, resp):
...
-                                        tags, None, project_name, owner_id)
+            datasource_item._set_values(id_, name, datasource_type, content_url, created_at, updated_at,
+                                        tags, None, project_name, owner_id, certified, certification_note)
             all_datasource_items.append(datasource_item)
         return all_datasource_items
 
     @staticmethod
     def _parse_element(datasource_xml):
-        id = datasource_xml.get('id', None)
+        blorp = datasource_xml
```
Comment (t8y8) [inline tableauserverclient/models/datasource_item.py]: oops, debug line

File: tableauserverclient/models/datasource_item.py [hunk 2]
```diff
@@ -104,22 +130,30 @@ def from_response(cls, resp):
...
+        id_ = datasource_xml.get('id', None)
         name = datasource_xml.get('name', None)
         datasource_type = datasource_xml.get('type', None)
         content_url = datasource_xml.get('contentUrl', None)
         created_at = parse_datetime(datasource_xml.get('createdAt', None))
         updated_at = parse_datetime(datasource_xml.get('updatedAt', None))
+        certification_note = datasource_xml.get('certificationNote', None)
```
Comment (LGraber) [inline tableauserverclient/models/datasource_item.py]: Can you have a certification note without being certified? Just curious?

Comment (t8y8) [inline tableauserverclient/models/datasource_item.py]: Technically yes. The UI may choose to not display them (yet) but it's not expressly forbidden. We may also add UI support for this anywho.

**Comment by RussTheAerialist** [comment]
🚀 🍦

File: tableauserverclient/server/request_factory.py [hunk 3]
```diff
@@ -65,6 +65,13 @@ def update_req(self, datasource_item):
         if datasource_item.owner_id:
             owner_element = ET.SubElement(datasource_element, 'owner')
             owner_element.attrib['id'] = datasource_item.owner_id
+
+        if datasource_item.certified:
```
Comment (t8y8) [inline tableauserverclient/server/request_factory.py]: wait, there's a bug. 

I think this won't unset

#rusty

**Pull request merged**